programs/pkey/pk_sign.c

```diff
@@ -134,7 +133,6 @@ int main( int argc, char *argv[] )
 
     if( ( f = fopen( filename, "wb+" ) ) == NULL )
     {
-        ret = 1;
```

Comment:
This error is necessary, as we are signalling that fopen has failed. Removing this assignment will result in the pk_sign binary returning 0, despite the fail.